next/lib/drupal/get-menus.ts

```diff
@@ -1,18 +1,23 @@
 import { GetStaticPropsContext } from "next";
+import pRetry from "p-retry";
 
 import { drupal } from "@/lib/drupal/drupal-client";
 
 import { MenuAvailable } from "../gql/graphql";
 import { GET_MENU } from "../graphql/queries";
 
 export async function getMenus({ locale }: GetStaticPropsContext) {
-  const [main, footer] = await Promise.all(
-    ["MAIN", "FOOTER"].map((menu) =>
-      drupal.doGraphQlRequest(GET_MENU, {
-        name: menu as MenuAvailable,
-        langcode: locale,
-      }),
-    ),
+  const [main, footer] = await pRetry(
+    () =>
+      Promise.all(
+        ["MAIN", "FOOTER"].map((menu) =>
+          drupal.doGraphQlRequest(GET_MENU, {
+            name: menu as MenuAvailable,
+            langcode: locale,
+          }),
+        ),
+      ),
+    { retries: 5 },
   );
```

Comment:
The thing with this one is, I think it'll retry if either of these `drupal.doGraphQlRequest`s fail?

If that's the case maybe it's better to structure it a bit differently so the pRetry deals with each promise individually and only retries whichever one failed? Something like this is what I'm thinking of (though I didn't try it):

```ts
const [main, footer] = await Promise.all(
  ["MAIN", "FOOTER"].map(menu =>
    pRetry(() => 
      drupal.doGraphQlRequest(GET_MENU, {
        name: menu,
        langcode: locale,
      }),
      { retries: 5 }
    )
  )
);
```

Reply:
ok! updated to your code (with a small addition) in 11999cfc066f1f2bef973a58705c997afe6399ec